express-notes-api/server.js

```diff
@@ -0,0 +1,127 @@
+import express from 'express';
+import * as fs from 'node:fs/promises';
+
+const app = express();
+
+const data = JSON.parse(await fs.readFile('data.json'));
+const notes = data.notes;
+
+app.get('/api/notes', (req, res) => {
+  console.log(req.method);
+  const array = [];
+  for (const id in notes) {
+    array.push(notes[id]);
+  }
+  res.status(200).json(array);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  console.log(req.method);
+  const id = Number(req.params.id);
+  let outcome = {};
+  outcome.error = `Cannot find note with id ${id}`;
```

Comment:
You could shorten this with
`const outcome = { error: `Cannot find note with id ${id}` };`